pkg/handler/handler.go

```diff
@@ -163,7 +175,7 @@ func (o *OpenAPIService) RegisterOpenAPIVersionedService(servePath string, handl
 						continue
 					}
 					// serve the first matching media type in the sorted clause list
-					result := accepts.GetDataAndEtag.Get()
+					result := o.getDataAndEtagFromCache(accepts.GetDataAndEtag)
```

Comment:
Ah, I think I know now how I committed this atrocity. I think I wanted to have the mutex NOT in the strcut, but directly in the RegisterOpenAPIVersionedService function. You basically only use the mutex in one place here so I think it's fine to just do have a local mutex?